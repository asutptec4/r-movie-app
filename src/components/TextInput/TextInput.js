import { useField } from 'formik';
import { string } from 'prop-types';
import React from 'react';

const TextInput = ({ label, ...props }) => {
  const [field, meta] = useField(props);
  return (
    <>
      <label className="app-form-label" htmlFor={props.id || props.name}>
        {label}
      </label>
      <input className="app-form-input" id={props.id || props.name} {...field} {...props} />
      {meta.touched && meta.error ? <div className="app-form-error">{meta.error}</div> : null}
    </>
  );
};

TextInput.propTypes = {
  label: string,
  id: string,
  name: string,
};

export default TextInput;
