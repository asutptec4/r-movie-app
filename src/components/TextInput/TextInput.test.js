import { render } from '@testing-library/react';
import { useField } from 'formik';
import React from 'react';

import TextInput from './TextInput';

jest.mock('formik');

describe('<TextInput />', () => {
  const defaultProps = { label: 'TextInputLabel' };

  test('should render props', () => {
    useField.mockReturnValue([{}, {}]);
    const { container } = render(<TextInput {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  test('should show error message igf touched', () => {
    const errorText = 'input error';
    useField.mockReturnValue([{}, { touched: true, error: errorText }]);
    const { getByText } = render(<TextInput {...defaultProps} />);
    getByText(errorText);
  });
});
