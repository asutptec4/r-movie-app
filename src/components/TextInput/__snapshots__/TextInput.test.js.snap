// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render props 1`] = `
<div>
  <label
    class="app-form-label"
  >
    TextInputLabel
  </label>
  <input
    class="app-form-input"
  />
</div>
`;
